### PR TITLE
users name schema

### PR DESCRIPTION
**Pull request by carelynntsai**
[Notion task](https://www.notion.so/uwblueprintexecs/users-name-schema-d08f5466736d4ac48d2dccbcc1e9e404)

## Changes
- removed "firstName" and "lastName" fields from user type and created a single field "name"
- updated seeders 
- updates user schema, model, resolvers
- created new migration

## Testing
- seeded data
- tested migration up and down functions
- tested resolvers

## Screenshots
image
image
image

### CODE REVIEW

**Comment by carelynntsai** [comment]
@petorsong wasn't sure if I should set the column to allowNull: false, the previous firstName/lastName columns weren't

File: migrations/20200719025533-add-name-column-users.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        "name",
+        {
+          type: Sequelize.DataTypes.STRING,
+        },
+        { transaction: t }
+      )
+      await queryInterface.sequelize.query(
+        `UPDATE users SET "name" = CONCAT("firstName", ' ', "lastName")`,
+        { transaction: t }
+      )
```
Comment (petorsong) [inline migrations/20200719025533-add-name-column-users.js:17]: i added a [migration that adds an admin user](https://github.com/uwblueprint/paramedics-web/blob/master/migrations/20200717012421-load-admin-user.js) before this change, which sets `lastName` to an empty string, which appends a space once this migration is run - this suggestion is to account for that edge case, but test it first cause i dunno if it's quite accurate

```suggestion
      await queryInterface.sequelize.query(
        `UPDATE users SET "name" = "firstName" WHERE "lastName"=''`,
        { transaction: t }
      );
      await queryInterface.sequelize.query(
        `UPDATE users SET "name" = CONCAT("firstName", ' ', "lastName") WHERE "lastName"<>''`,
        { transaction: t }
      );
```

File: migrations/20200719025533-add-name-column-users.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction(async (t) => {
```
Comment (petorsong) [inline migrations/20200719025533-add-name-column-users.js:5]: is there a particular need to use this await/async syntax instead of the `Promise.all` syntax?
https://sequelize.org/master/manual/migrations.html#migration-skeleton

File: migrations/20200719025533-add-name-column-users.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      await queryInterface.addColumn(
+        "users",
+        "lastName",
+        {
+          type: Sequelize.DataTypes.STRING,
+        },
+        { transaction: t }
+      );
+      await queryInterface.sequelize.query(
+        'UPDATE users SET "firstName" = "name", "lastName" = "name"',
```
Comment (petorsong) [inline migrations/20200719025533-add-name-column-users.js:42]: this isn't super sustainable - the names get too long if you run undo a bunch of times (like i did during testing); if you can find an easy way to do a slightly different operation (e.g. explode `name` on ' ' and put the first part in `firstName` and second part in `lastName`) that would be preferred, otherwise just lemme know and we'll flag it to be fixed later

Comment (carelynntsai) [reply migrations/20200719025533-add-name-column-users.js:46]: would it be better to do a similar thing like you did with admin and ``UPDATE users SET "firstName" = "name", "lastName" = ''``

Comment (petorsong) [reply migrations/20200719025533-add-name-column-users.js:46]: let's just leave it as is, but maybe add a comment here to say running this down migration too often may result in issues
